### ISSUE TITLE
DDP GPU utilization problem

### ISSUE THREAD

**Issue by dragondx** [PyTorchLightning/pytorch-lightning#11157]
### Discussed in https://github.com/PyTorchLightning/pytorch-lightning/discussions/10670

<div type='discussions-op-text'>

<sup>Originally posted by **dragondx** November 22, 2021</sup>
Training MLP using DDP with 2gpus. Pretty standard code.  
We observe an effect where the training speed slows down over time.  At initial iterations, we get 1s/it. Both GPUs gets 100% utilization with occasional low utilization presumably it is doing gathering/backprop ops. As training goes on for a few thousands iterations, we get 2s/it.  One gpu get 100% utilization consistently (the master?), the other gpu waits a long time with low utilization (1-2%) before getting short burst of 100% utilization.  We also noticed that CPU utilization (80% along all cores) tends to be much higher at earlier iterations.  After a few thousand iterations, CPU utilization is low (20%, with occasional 90% in some cores).  Is this some sort of timing problem for dataloading?  There is no preprocessing for CPU other than reading data from disk.  

We use a custom dataloader for our use case, since we have a lot of data.  Using numpy memmap to get the datapoint.
Dataloder params:
torch.utils.data.DataLoader(train_dataset, batch_size=batch_size, shuffle=True, persistent_workers=True, pin_memory=True, num_workers=16, prefetch_factor=128)

We wonder what is causing this.  Any help will be greatly appreciated.</div>

Update: We changed to local training using fast nvme SSDs, training speed still slows down as the epoch goes on.  Is this something related to "cache" or something accumulated by pytorch lightning?  We did not allocate any dangling variables in the model.

**Comment by awaelchli** [comment]
Your prefetch factor is unusually high. You are prefetching 16 * 128 samples. This explains your observation for the CPU core utilization, right? How is your CPU memory utilization?

**Comment by dragondx** [comment]
Thanks for the reply.
CPU memory usage is not really high 80 GB to 100GB. Never completely utilized. If I use a smaller perfetch factor, each iteration will take a long time. That is, even when using a nvme ssd.
